src/log.cpp

```diff
@@ -0,0 +1,79 @@
+#include "log.hpp"
+
+Log::Log(std::string path)
+{
+	this->path = path;
+	this->logMsg.open(path, std::ofstream::out | std::ofstream::app);
+	ASSERT(this->logMsg.is_open(), "could not open log file at %s",
+	       path.c_str());
+}
+
+std::ofstream& Log::getStream()
+{
+	return this->logMsg;
+}
+
+void Log::close()
+{
+	this->logMsg.close();
+}
+
+void Log::TCPconnection()
+{
+	this->getStream() << "[Time]: Peer [peer_ID 1] makes a connection to Peer [peer_ID 2].\n";
+	logMsg.close();

```

Comment:
We need a better way to handle closing/opening the file stream, but this is good enough for the time being